core/math/transform_3d.h

```diff
@@ -131,15 +131,18 @@ _FORCE_INLINE_ Vector3 Transform3D::xform_inv(const Vector3 &p_vector) const {
 }
 
 _FORCE_INLINE_ Plane Transform3D::xform(const Plane &p_plane) const {
+	// transform a single point on the plane
 	Vector3 point = p_plane.normal * p_plane.d;
-	Vector3 point_dir = point + p_plane.normal;
 	point = xform(point);
-	point_dir = xform(point_dir);
 
-	Vector3 normal = point_dir - point;
+	// use inverse transpose for correct normals with non-uniform scaling
+	Basis b = basis;
+	b.invert();
+	b.transpose();
+	Vector3 normal = b.xform(p_plane.normal);
```

Comment:
I haven't checked the math, but maybe `xform_inv` is actually a faster shortcut to avoid computing the full inverse transpose matrix?

Reply:
It is, but I believe it isn't as robust as doing proper inverse transpose.

Reply:
Just confirmed, `xform_inv` doesn't work for Vector3 with non-uniform scales, so I'll change that.